applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -355,33 +363,40 @@ class ExplicitCentralDifferencesScheme
     /**
      * @brief This method initializes some rutines related with the explicit scheme
      * @param rModelPart The model of the problem to solve
+     * @param DoF The current dimention of the problem
      */
-    void InitializeExplicitScheme(ModelPart& rModelPart)
+    void InitializeExplicitScheme(
+        ModelPart& rModelPart,
+        const SizeType DoF = 3
+        )
     {
         KRATOS_TRY
 
         /// The array of ndoes
         NodesArrayType& r_nodes = rModelPart.Nodes();
 
-        /// Working in 3D always by default
-        const SizeType dim(3);
-
         // The first iterator of the array of nodes
-        auto it_begin = rModelPart.NodesBegin();
+        const auto it_begin = rModelPart.NodesBegin();
 
         /// Initialise the database of the nodes
-        const array_1d<double, 3> zero_array(3, 0.0);
-        #pragma omp parallel for firstprivate(it_begin)
+        const array_1d<double, 3> zero_array = ZeroVector(3);
+        #pragma omp parallel for
         for (int i = 0; i < static_cast<int>(r_nodes.size()); ++i) {
             auto it_node = (it_begin + i);
             it_node->SetValue(NODAL_MASS, 0.0);
             it_node->SetValue(MIDDLE_VELOCITY, zero_array);
-            it_node->SetValue(MIDDLE_ANGULAR_VELOCITY, zero_array);
-            it_node->SetValue(NODAL_INERTIA, zero_array);
         }
-
         const bool has_dof_for_rot_z = it_begin->HasDofFor(ROTATION_Z);
-        #pragma omp parallel for firstprivate(it_begin)
+        if (has_dof_for_rot_z) {
+            #pragma omp parallel for
+            for (int i = 0; i < static_cast<int>(r_nodes.size()); ++i) {
+                auto it_node = (it_begin + i);
+                it_node->SetValue(MIDDLE_ANGULAR_VELOCITY, zero_array);
+                it_node->SetValue(NODAL_INERTIA, zero_array);
```

Comment:
I am currently discussing with @roigcarlo if this is save in omp or not => #3712

Reply:
it is safe. on the contrary if you used node->GetValue(...) = ... it would throw an error if uninitialized